### ISSUE TITLE
Fine-tuning the pretrained weight

### ISSUE THREAD

**Issue by jesseisoaho** [roboflow/rf-detr#224]
Hi 
I am trying to fine-tune the base rf-detr with my dataset that has 1 class
I use this
```py
from rfdetr import RFDETRBase

model = RFDETRBase(num_classes=1,pretrained=True,)

print(model.model_config)

model.train(
    dataset_dir      = "/workspace/storage1/jesse/coco_dataset/bolt_plane_module_coco",
    train_ann_file   = "/workspace/storage1/jesse/coco_dataset/bolt_plane_module_coco/train/_annotations.coco.json",
    val_ann_file     = "/workspace/storage1/jesse/coco_dataset/bolt_plane_module_coco/valid/_annotations.coco.json",
    output_dir       = "/workspace/sbtc-cv/segmentation/rt_detr/out_plane_module",
    epochs=120,                  
    batch_size=32,                
    grad_accum_steps=1,             
    lr=1e-4,                          
    lr_encoder=1e-5,                   
    resolution=672,              
    weight_decay=1e-4,
    device="cuda",
    use_ema=True,                    
    checkpoint_interval=1,
    tensorboard=True,
    early_stopping=True,
    early_stopping_patience=7,
    early_stopping_min_delta=0.001,
    early_stopping_use_ema=True,
    amp=True                         
)
```
But with this I get 
Loading pretrain weights
num_classes mismatch: pretrain weights has 90 classes, but your model has 1 classes
reinitializing detection head with 90 classes

How should I do this to not make the rf-detr load the basic coco classes

**Comment by anyaflya** [comment]
Hello, 

I brought up this issue at [issue #51](https://github.com/roboflow/rf-detr/issues/51). Short answer: do not instantiate your model with the number of classes.

**Comment by Borda** [comment]
✅ This has been fixed in v1.6.0! Key changes:

- **Detection head reinitialization** no longer overwrites fine-tuned weights
- **`class_names`** now correctly reflect your dataset labels after training
- Misleading `num_classes` warnings were fixed in v1.4.2
- v1.6.0 also added composable Lightning training with proper `num_classes` handling

To upgrade: `pip install rfdetr>=1.6.1`

For the canonical fine-tuning guide, see #878. Closing as resolved — feel free to reopen if the issue persists on v1.6.1!
